### PR TITLE
Added dog review form

### PR DESCRIPTION
**Pull request by benlimer**
Created dog review form and necessary post requests to the backend
Updated dogsRouter
Implemented user session functionality to correctly link reviews with userId 
Refactored post methods using Fetcher function

### CODE REVIEW

File: client/src/components/App.js
Comment (dovakeen118) [inline client/src/components/App.js]: If you are trying to pass the `currentUser` and still have access to things like `props.match`, we should instead:

Wrap the component in a route
```js
<Route exact path="/dogs/:id">
  <DogShowPage user={currentUser} />
</Route>
```

Then in the show component, you'll make use of `withRouter`
```js
// /DogShow.js

import { withRoute } from "react-router"

...

export default withRouter(DogShow)
```

Then, within the component you can use `props.match`.etc as needed!

File: client/src/components/layout/DogReviewForm.js
Comment (dovakeen118) [inline client/src/components/layout/DogReviewForm.js]: This outer div is not used and is clutter in the DOM

File: client/src/components/layout/DogShowPage.js
Comment (dovakeen118) [inline client/src/components/layout/DogShowPage.js]: empty div!

File: client/src/components/layout/DogShowPage.js
Comment (dovakeen118) [inline client/src/components/layout/DogShowPage.js:9]: Where do these reviews come from? As in it doesn't look like we are displaying the reviews anywhere on the page? Is that a separate feature?

We would also assume that the reviews would come from the getDog fetch request, for the reviews about that dog.

File: server/src/db/migrations/20220201192529_createReviews.cjs
Comment (dovakeen118) [inline server/src/db/migrations/20220201192529_createReviews.cjs:14]: Careful making changes to existing migrations that have been merged in! This change will need to be communicated to the group to let everyone know that the database schema will now be altered going forward if this is merged in. This change would be affecting the existing migrations on everyone else's machine. When making database changes, we should ideally do so in new migrations so that there is a clear history of changes against the database.

**Review by dovakeen118 — CHANGES_REQUESTED** [review]
Great work! Some comments for clean up/refactors

File: server/src/db/migrations/20220201192529_createReviews.cjs
Comment (benlimer) [reply server/src/db/migrations/20220201192529_createReviews.cjs:14]: This was a group decision to delete straight off of the migration file. As we are correcting something that shouldn't have been there, not adding something in. we thought it was more of a system design issue.

File: client/src/components/layout/DogShowPage.js
Comment (benlimer) [reply client/src/components/layout/DogShowPage.js:9]: We are not displaying these reviews because we are waiting for the other group to finish on "listing existing reviews" on the right side of the show page. So that we can add to that list. We will add the functionality to add the new reviews to the list once we merge today

File: client/src/components/authentication/AuthenticatedRoute.js
Comment (dovakeen118) [inline client/src/components/authentication/AuthenticatedRoute.js]: Looks like merge comments were left in the PR, along with some duplicated info

File: client/src/components/authentication/AuthenticatedRoute.js
Comment (dovakeen118) [inline client/src/components/authentication/AuthenticatedRoute.js]: We can remove that additional empty space that was added here. We can have one empty space before a closing tag `...} />`

File: client/src/components/layout/DogReviewForm.js
Comment (dovakeen118) [inline client/src/components/layout/DogReviewForm.js:44]: Woah, let's clean up the indentation here and all these extra, empty spaces

File: client/src/components/layout/services/Fetcher.js
Comment (dovakeen118) [inline client/src/components/layout/services/Fetcher.js]: Some merge conflicts here as well that were resolved a little wonky

File: client/src/components/layout/services/Fetcher.js
Comment (dovakeen118) [inline client/src/components/layout/services/Fetcher.js]: Here as well

File: server/src/routes/api/v1/dogsRouter.js
Comment (dovakeen118) [inline server/src/routes/api/v1/dogsRouter.js]: I'm not sure what the difference is between this endpoint and the POST one above it? I see - this endpoint is for creating a review for a dog. As such, this should instead be a nested router. We are not interacting with the `dog` resource as this router would suggest. but instead want to make a `dogReviews` nested router for better organization for this specific POST reviews endpoint. With that, we want to make sure we're following RESTful API endpoints, so the POST path would be `/api/v1/dogs/:dogId/reviews` since it is the _reviews_ resource we're trying to create (not a dog, or dogId)

**Review by dovakeen118 — CHANGES_REQUESTED** [review]
Looking good! Some clean up to do from when the merge conflicts were resolve a little wonky. There are also some comments for refactoring some best practices

**Pull request merged**
